File: 0793. Preimage Size of Factorial Zeroes Function/Solution.c
long long numOfTrailingZeros(long long x)
{
    long long res = 0;
    for (; x; x /= 5)
        res += x / 5;
    return res;
}

int preimageSizeFZF(int k)
{
    long long l = 0, r = 5LL * k;
    while (l <= r) {
        long long m = l + ((r - l) >> 1);
        int zeros = numOfTrailingZeros(m);

        if (zeros == k)
            return 5;
        else if (zeros < k)
            l = m + 1;
        else
            r = m - 1;
    }
    return 0;
}

File: 0793. Preimage Size of Factorial Zeroes Function/Solution.py
class Solution:
    def preimageSizeFZF(self, k: int) -> int:

        def numOfTrailingZeros(x: int) -> int:
            res = 0
            while x:
                res += x // 5
                x //= 5
            return res

        l, r = 0, 5 * k
        while l <= r:
            m = l + ((r - l) >> 1)
            zeros = numOfTrailingZeros(m)
            if zeros < k:
                l = m + 1
            elif zeros > k:
                r = m - 1
            else:
                return 5
        return 0
